entities/moduletools/object_diff_test.go

```diff
@@ -0,0 +1,102 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package moduletools
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestObjectDiff(t *testing.T) {
+	t.Run("strings are the same", func(t *testing.T) {
+		objDiff := NewObjectDiff(nil).
+			WithProp("sameStrings", "Some string", "Some string")
+
+		assert.False(t, objDiff.IsChangedProp("sameStrings"))
+	})
+
+	t.Run("strings are different", func(t *testing.T) {
+		objDiff := NewObjectDiff(nil).
+			WithProp("differentStrings1", "Some string", "some string").
+			WithProp("differentStrings2", "Some string", "Some string, but different")
+
+		assert.True(t, objDiff.IsChangedProp("differentStrings1"))
+		assert.True(t, objDiff.IsChangedProp("differentStrings2"))
+	})
+
+	t.Run("string slices are the same", func(t *testing.T) {
+		objDiff := NewObjectDiff(nil).
+			WithProp("sameStringSlices1", []string{"aa", "bb", "cc"}, []string{"aa", "bb", "cc"}).
```

Comment:
The unit test is not covering cases where we have a mixture of []string, []interface{}

For example try something like 
WithProp("sameStringSlices3", []string{"aa", "bb", "cc"}, []interface{}{"aa", "bb", "cc"})
assert.False(t, objDiff.IsChangedProp("sameStringSlices3"))

Reply:
Right, I forgot about that case as `[]interface{}` support was added after tested were created. Will add them.

Reply:
more usecases added